ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/forms/ReportParametersForm.tsx

```diff
@@ -32,54 +31,34 @@ import CollectionSelection from './CollectionSelection';
 
 export type ReportParametersFormParams = {
     formValues: ReportFormValues;
-    setFormValues: SetReportFormValues;
+    setFormFieldValue: SetReportFormFieldValue;
 };
 
 function ReportParametersForm({
     formValues,
-    setFormValues,
+    setFormFieldValue,
 }: ReportParametersFormParams): ReactElement {
     const { hasReadWriteAccess } = usePermissions();
     const canWriteCollections = hasReadWriteAccess('WorkflowAdministration');
 
     const handleTextChange = (fieldName: string) => (value: string) => {
-        setFormValues((prevValues) => {
-            const newValues = { ...prevValues };
-            set(newValues, fieldName, value);
-            return newValues;
-        });
+        setFormFieldValue(fieldName, value);
     };
 
     const handleSelectChange = (name: string, value: string) => {
-        setFormValues((prevValues) => {
-            const newValues = { ...prevValues };
-            set(newValues, name, value);
-            return newValues;
-        });
+        setFormFieldValue(name, value);
     };
 
     const handleCheckboxSelectChange = (fieldName: string) => (selection: string[]) => {
-        setFormValues((prevValues) => {
-            const newValues = { ...prevValues };
-            set(newValues, fieldName, selection);
-            return newValues;
-        });
+        setFormFieldValue(fieldName, selection);
     };
 
-    const handleDateSelection = (fieldName: string) => (_event, str) => {
-        setFormValues((prevValues) => {
-            const newValues = { ...prevValues };
-            set(newValues, fieldName, str);
-            return newValues;
-        });
+    const handleDateSelection = (fieldName: string) => (_event, selection) => {
+        setFormFieldValue(fieldName, selection);
```

Comment:
I think PF 5 (not out yet) will standardize the order of change handler params, so we may be able to simplify and reuse some of these similar handlers.

Even now, could `handleSelectChange` be used for 
- handleCheckboxSelectChange
- handleCollectionSelection

Reply:
I don't think we can do that since the handlers are still pretty different. We'd have to standardize our own components to work in a similar format. Right now they don't